src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -714,12 +714,16 @@ async def _schedule_single_node_session(
             raise GenericBadRequest(
                 "Cannot assign multiple kernels with different architectures' single node session",
             )
+        if requested_architectures is None:
+            raise GenericBadRequest("Requested session has no architecture information")
```

Comment:
This cannot happen because `requested_architectures` is always `set`, not `None`